aiohttp/web_fileresponse.py

```diff
@@ -121,19 +121,31 @@ async def _precondition_failed(
         self.content_length = 0
         return await super().prepare(request)
 
-    async def prepare(self, request: "BaseRequest") -> Optional[AbstractStreamWriter]:
+    def _get_file_path_stat_and_gzip(
+        self, check_for_gzipped_file: bool
+    ) -> Tuple[pathlib.Path, os.stat_result, bool]:
+        """Return the file path, stat result, and gzip status.
+
+        This method should be called from a thread executor
+        since it calls os.stat which may block.
+        """
         filepath = self._path
-
-        gzip = False
-        if "gzip" in request.headers.get(hdrs.ACCEPT_ENCODING, ""):
+        if check_for_gzipped_file:
             gzip_path = filepath.with_name(filepath.name + ".gz")
+            try:
```

Comment:
Personally, I'd prefer a suppress() rather than having a pass statement.

Reply:
Usually I avoid `suppress` any time it might be performance sensitive since its much slower than a `try` pass

Reply:
OK, never really considered there might be a performance difference..